docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -27,644 +27,348 @@ under the License.
-    The `DescribeStreamConsumer` operation has a limit of 20 [transactions per second](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_DescribeStreamConsumer.html),
-    this means application startup time will increase by roughly `parallelism/20 seconds`.
-  - `EAGER`: Stream consumers are registered in the `FlinkKinesisConsumer` constructor.
-    If the stream consumer already exists, it will be reused. 
-    This will result in registration occurring when the job is constructed, 
-    either on the Flink Job Manager or client environment submitting the job.
-    Using this strategy results in a single thread registering and retrieving the stream consumer ARN, 
-    reducing startup time over `LAZY` (with large parallelism).
-    However, consider that the client environment will require access to the AWS services.
-  - `NONE`: Stream consumer registration is not performed by `FlinkKinesisConsumer`.
-    Registration must be performed externally using the [AWS CLI or SDK](https://aws.amazon.com/tools/)
-    to invoke [RegisterStreamConsumer](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_RegisterStreamConsumer.html).
-    Stream consumer ARNs should be provided to the job via the consumer configuration.
-- Deregistration
-  - `LAZY` (default): Stream consumers are deregistered when the job is shutdown gracefully.
-    In the event that a job terminates without executing the shutdown hooks, stream consumers will remain active.
-    In this situation the stream consumers will be gracefully reused when the application restarts. 
-  - `EAGER|NONE`: Stream consumer deregistration is not performed by `FlinkKinesisConsumer`.
-
-Below is an example configuration to use the `EAGER` registration strategy:
-
-{{< tabs "a85d716b-6c1c-46d8-9ee4-12d8380a0c06" >}}
-{{< tab "Java" >}}
-```java
-Properties consumerConfig = new Properties();
-consumerConfig.put(AWSConfigConstants.AWS_REGION, "us-east-1");
-consumerConfig.put(ConsumerConfigConstants.STREAM_INITIAL_POSITION, "LATEST");
-
-consumerConfig.put(ConsumerConfigConstants.RECORD_PUBLISHER_TYPE, 
-    ConsumerConfigConstants.RecordPublisherType.EFO.name());
-consumerConfig.put(ConsumerConfigConstants.EFO_CONSUMER_NAME, "my-flink-efo-consumer");
+### Parallelism and Number of Shards
+The configured parallelism of the `KinesisStreamsSource` is independent of the total number of shards in the Kinesis streams.
+- If the parallelism of the `KinesisStreamsSource` is less than the total number of shards, then a single parallel subtask would handle multiple shards.
+- If the parallelism of the `KinesisStreamsSource` is more than the total number of shards, then there will be some parallel subtasks that do not read from any shards. If this is the case, users will need to set up `withIdleness` on the `WatermarkStrategy`. Failing to do so will mean watermark generation will get blocked due to idle subtasks.
```

Comment:
Is this also true if you specify `WatermarkStrategy.noWatermarks()` ?

Reply:
If there are no watermarks, the watermark generation cannot be blocked!